src/NServiceBus.Core/Routing/RoutingComponent.cs

```diff
@@ -43,16 +54,22 @@ public static RoutingComponent Initialize(Configuration configuration, Transport
             {
                 distributionPolicy.SetDistributionStrategy(distributionStrategy);
             }
-            
+
             configuration.ConfiguredUnicastRoutes?.Apply(unicastRoutingTable, conventions);
 
+            pipelineSettings.Register(new ApplyReplyToAddressBehavior(
+                    receiveConfiguration.LocalAddress,
+                    receiveConfiguration.InstanceSpecificQueue,
+                    configuration.PublicReturnAddress),
+                "Applies the public reply to address to outgoing messages");
+
             var sendRouter = new UnicastSendRouter(
-                receiveConfiguration == null, 
-                receiveConfiguration?.QueueNameBase, 
-                receiveConfiguration?.InstanceSpecificQueue, 
-                distributionPolicy, 
-                unicastRoutingTable, 
-                endpointInstances, 
+                configuration.IsSendOnly,
```

Comment:
@andreasohlund this is a temporary workaround. Related to the question about where the `"endpoint.sendonly"` belongs.

Reply:
nm. It was my confusion. `ReceiveConfiguration` behaves the same as it did before. PEBCAK 🙂